CMakeLists.txt

```diff
@@ -30,8 +30,13 @@ message(STATUS "Using CMake ${CMAKE_VERSION}")
 
 # Enforce the C++ standard we are using and turn off compiler-specific extensions
 set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_STANDARD_REQUIRED On)
-set(CMAKE_CXX_EXTENSIONS Off)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+# We do not use C++ modules (yet). Turn off scanning to avoid issues with clang-tidy.
+# If you want to add module support, this problem needs to be revisited. Our hope
+# is that CMake and/or clang-tidy will be updated to handle modules better.
+set(CMAKE_CXX_SCAN_FOR_MODULES OFF)
```

Comment:
This fixes clang-tidy, which, funnily enough, does not understand the modmap files emitted by clang. I am pretty sure this is already fixed or will be fixed in newer versions. We do not have module support anyway.